### ISSUE TITLE
pick on transformed rectilinear grids doesn't work

### ISSUE THREAD

**Issue by aowen87** [aowen87/BAR#2175]
If you open a file with a transformed rectilinear grid, picking on it simply reports that there were no intersections.  

For example, open trunk/data/CHGCAR (it's a sub-format of the VASP reader, but it should choose VASP automatically these days), make a pseudocolor plot, and try zone or node pick, and for the most part it doesn't report any 

I thought I'd made this work before.  If so, I believe it would have been possible to create a pick test case for it, but I don't believe I did.  It would be really smart to add one when we fix this.

If this *didn't* work, then we should probably add support for it, though the priority isn't the same if it's not something that ever worked.

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. As such, not all 
information was able to be captured in the transition. Below is
a complete record of the original redmine ticket.

Ticket number: 593
Status: Resolved
Project: VisIt
Tracker: Bug
Priority: High
Subject: pick on transformed rectilinear grids doesn't work
Assigned to: Kathleen Biagas
Category: 
Target version: 2.3
Author: Jeremy Meredith
Start: 02/09/2011
Due date: 
% Done: 0
Estimated time: 2.0
Created: 02/09/2011 03:04 pm
Updated: 02/24/2011 03:20 pm
Likelihood: 3 - Occasional
Severity: 2 - Minor Irritation
Found in version: 2.0.1
Impact: 
Expected Use: 
OS: All
Support Group: Any
Description:
 If you open a file with a transformed rectilinear grid, picking on it simply reports that there were no intersections.  

For example, open trunk/data/CHGCAR (it's a sub-format of the VASP reader, but it should choose VASP automatically these days), make a pseudocolor plot, and try zone or node pick, and for the most part it doesn't report any 

I thought I'd made this work before.  If so, I believe it would have been possible to create a pick test case for it, but I don't believe I did.  It would be really smart to add one when we fix this.

If this *didn't* work, then we should probably add support for it, though the priority isn't the same if it's not something that ever worked.

Comments:
I'd forgotten that there's a workaround: add the ExternalSurface operator, since this forces it back into the world coordinate system before rendering.  Plus, you really only need it for, e.g. Pseudocolor plots; the Contour plot you can already pick on, and you can't pick on Volume even for other mesh types.  It may still be that there's an easy fix to make this work without such workarounds, but if there's a real obstacle in fixing this, we might take the existence of a workaround into consideration.
I don't think this ever worked.Utilizing the rectilinearGridTransform on the ray Points in ViewerQueryManager can get intersections.However, preliminary testing with this addition shows Pick returning different results in parallel than in serial.
Note that the CHGCAR format supports automatic domain decomposition, so depending on what you tested, different parallel results may be perfectly acceptable, depending on what you're looking at.  So that path may be fine -- if you'd like help testing or debugging, I'll gladly volunteer.
Pick needed to take into account the rectilinear grid transformation matrix.SVN revision 14154 (trunk)M      avt/Queries/Pick/avtZonePickQuery.CM      avt/Queries/Pick/avtNodePickQuery.CM      viewer/main/ViewerQueryManager.CThe differing parallel/serial results is due to the format performing automatic domain decomposition.A new ticket will be submitted regarding this.
